packages/runtime/test/src/interpreter/interpreter.ts

```diff
@@ -0,0 +1,126 @@
+
+	const reset = (): void => {
+		while (interpreter.stack.length() !== 0) {
+			interpreter.stack.pop();
+		}
+		interpreter.currentInnerTxnGroup = [];
+		interpreter.runtime.ctx.pooledApplCost = 0;
+		interpreter.instructions = [];
+		interpreter.innerTxnGroups = [];
+		interpreter.instructionIndex = 0;
+		interpreter.runtime.ctx.tx = { ...TXN_OBJ, snd: Buffer.from(elonPk) };
+		interpreter.runtime.ctx.gtxs = [interpreter.runtime.ctx.tx];
+		interpreter.runtime.ctx.isInnerTx = false;
+		// set new tx receipt
+		interpreter.runtime.ctx.state.txReceipts.set(TXN_OBJ.txID, {
+			txn: TXN_OBJ,
+			txID: TXN_OBJ.txID,
+		});
+	};
+
+	const setUpInterpreter = (
+		tealVersion: number,
+		appBalance: number = ALGORAND_ACCOUNT_MIN_BALANCE
+	): void => {
+		// setup 1st account (to be used as sender)
+		elonAcc = new AccountStore(0, elonMuskAccount); // setup test account
+		setDummyAccInfo(elonAcc);
+
+		// setup 2nd account
+		johnAcc = new AccountStore(0, johnAccount);
+
+		// setup 2nd account
+		bobAccount = new AccountStore(1000000, bobAcc);
+
+		// setup application account
+		appAccAddr = getApplicationAddress(TXN_OBJ.apid);
+		applicationAccount = new AccountStore(appBalance, {
+			addr: appAccAddr,
+			sk: new Uint8Array(0),
+		});
+
+		interpreter = new Interpreter();
+		interpreter.runtime = new Runtime([elonAcc, johnAcc, bobAccount, applicationAccount]);
+		interpreter.tealVersion = tealVersion;
+		reset();
+	};
+	this.beforeAll(() => {
+		setUpInterpreter(6); //setup interpreter for execute
+	});
+
+	describe("Teal cost", () => {
+		useFixture("teal-files");
+		beforeEach(() => {
+			reset(); //reset the state of interpreter
+		});
+		it("Should return correct cost for a .teal program", () => {
+			let file = "test-file-1.teal";
+			interpreter.execute(getProgram(file), ExecutionMode.SIGNATURE, interpreter.runtime);
+			assert.equal(interpreter.cost, 3);
+
+			reset(); //after each execute need to reset interpreter
+			interpreter.cost = 0;
+			file = "test-file-3.teal";
+			interpreter.execute(getProgram(file), ExecutionMode.SIGNATURE, interpreter.runtime);
+			assert.equal(interpreter.cost, 3);
+
+			reset();
+			interpreter.cost = 0;
+			file = "test-interpreter.teal";
+			interpreter.execute(getProgram(file), ExecutionMode.SIGNATURE, interpreter.runtime);
+			assert.equal(interpreter.cost, 5);
+
+			reset();
+			interpreter.cost = 0;
+			file = "test-if-else.teal";
+			interpreter.execute(getProgram(file), ExecutionMode.SIGNATURE, interpreter.runtime);
+			assert.equal(interpreter.cost, 9);
+		});
+
+		it("Should fail when executing wrong logic teal", () => {
+			// logic of teal file failed
+			reset();
+			interpreter.cost = 0;
+			const file = "test-label.teal";
+			expectRuntimeError(
+				() =>
+					interpreter.execute(getProgram(file), ExecutionMode.SIGNATURE, interpreter.runtime),
+				RUNTIME_ERRORS.TEAL.REJECTED_BY_LOGIC
+			);
+		});
+	});
+
+	// it("Should return correct cost for a .teal program", () => {
```

Comment:
Why is it commented out?

Reply:
Oh sorry, this one I test yesterday and I find that this file is logic failed. 
Let me delete this

Reply:
But should it fail? If the .teal file is correct but it fails we should take a closer look to it

Reply:
It is like the file teal "test-label.teal" does not have any thing to return and maybe it is wrong logic.
But I find that code like this::

`#pragma version 4\n
int 5\n
int 3\n
==`

would cause the error TEAL.REJECTED_BY_LOGIC